truss-chains/examples/rag/rag_chain.py

```diff
@@ -0,0 +1,144 @@
+            He also has a passion for cooking and often experiments with new recipes in his spare time.
+            """,
+            """
+            Maria Silva is a nurse residing in New York City. She is dedicated to providing
+            compassionate care to her patients. Maria finds joy in gardening and often
+            spends her weekends tending to her vibrant flower beds and vegetable garden.
+            """,
+            """
+            John Smith is a journalist based in San Francisco. He reports on international politics
+            and has a knack for uncovering compelling stories. Outside of work, John is a history
+            buff who enjoys visiting museums and historical sites.
+            """,
+            """
+            Aisha Mohammed lives in Tokyo and works as a graphic designer. She creates visually stunning
+            graphics for a variety of clients. Aisha loves to paint and often showcases her
+            artwork in local exhibitions.
+            """,
+            """
+            Carlos Mendes is an environmental engineer in San Francisco.
+            He is passionate about developing sustainable solutions for urban areas.
+            In his leisure time, Carlos enjoys surfing and participating in beach clean-up initiatives.
+            """,
+        ]
+
+        self._collection.add(documents=documents, ids=[f"id{n}" for n in range(10)])
+
+    async def run_remote(self, query: str) -> list[str]:
+        results = self._collection.query(
+            query_texts=[query],  # Chroma will embed this for you
+            n_results=2,  # how many results to return
+        )
+
+        if results is None or not results:
+            raise ValueError("No bios returned from the query")
+
+        if not results["documents"] or not results["documents"][0]:
+            raise ValueError("Bios are empty")
+
+        return results["documents"][0]
+
+
+class LLMClient(chains.StubBase):
+    async def run_remote(self, new_bio: str, bios: list[str]) -> str:
+        prompt = f"""
+        You are matching alumni of a college to help them make connections.
+        Explain why the person described first would want to meet the people
+        selected from the matching database.
+        Person you're matching: {new_bio}
+        People from database: {" ".join(bios)}
+        """
+        resp = await self._remote.predict_async(
+            json_payload={
+                "messages": [{"role": "user", "content": prompt}],
+                "stream": False,
+            }
+        )
+        return resp["output"]
+
+
+@chains.mark_entrypoint
+class RAG(chains.ChainletBase):
+    def __init__(
+        self,
+        vector_store: VectorStore = chains.depends(VectorStore),
+        context: chains.DeploymentContext = chains.depends_context(),
+    ):
+        self._vector_store = vector_store
+        self._llm = LLMClient.from_url(
+            "https://model-03ym4ev3.api.baseten.co/production/predict", context
```

Comment:
To indicate that users need to update this URL, it would be better to make this a file level constant and comment next to it, that it must be updated.